pkg/config/msp.go

```diff
@@ -670,6 +670,24 @@ func (m *MSP) validateCACerts() error {
 	if err != nil {
 		return fmt.Errorf("invalid intermediate cert: %v", err)
 	}
+	rootCertPool := x509.NewCertPool()
+	for _, cert := range m.RootCerts {
+		rootCertPool.AddCert(cert)
+	}
+	//TODO: follow the workaround that msp code use to incorporate cert.Verify()
+	for _, ic := range m.IntermediateCerts {
+		validIntermediateCert := false
+		for _, rc := range m.RootCerts {
+			err := ic.CheckSignatureFrom(rc)
+			if err == nil {
+				validIntermediateCert = true
+				break
+			}
+		}
+		if !validIntermediateCert {
+			return errors.New("invalid intermediate cert")
```

Comment:
I know this error message may not be long-lived, but just saying that the intermediate cert is invalid doesn't really provide the user any actionable item to correct things. This should really state that the intermediate cert has not been signed by any of the root certs (and should probably include the cert's serial number). Something like: "intermediate cert not signed by any root certs of this MSP. serial number: 123456".